client/app/scripts/superdesk/editor/editor.js

```diff
@@ -725,11 +1107,74 @@ angular.module('superdesk.editor', ['superdesk.editor.spellcheck'])
 
                     scope.cursor = {};
                     render(null, null, true);
+
+                    function extractBlockContentsFromCaret() {
+                        function getBlockContainer(node) {
+                            while (node) {
+                                if (node.nodeType === 1 && /^(P|H[1-6]|DIV)$/i.test(node.nodeName)) {
+                                    return node;
+                                }
+                                node = node.parentNode;
+                            }
+                        }
+                        var sel = window.getSelection();
+                        if (sel.rangeCount) {
+                            var selRange = sel.getRangeAt(0);
+                            var blockEl = getBlockContainer(selRange.endContainer);
+                            if (blockEl) {
+                                var range = selRange.cloneRange();
+                                range.selectNodeContents(blockEl);
+                                range.setStart(selRange.endContainer, selRange.endOffset);
+                                return range.extractContents();
+                            }
+                        }
+                    }
+
+                    // Actions to support multi blocks edition
+                    if (scope.config.multiBlockEdition) {
+                        editorElem.on('keyup', function(e) {
+                            $timeout(function () {
+                                // press enter, create a new block
+                                 if (e.keyCode === 13) {
```

Comment:
Despite the comment, I think it still makes sense to define a key map for even better readability, e.g.
```js
if (e.keyCode === KEYS.ENTER) {...
```